blockchain/chain.go

```diff
@@ -1304,15 +1304,6 @@ type Config struct {
 	// time is adjusted to be in agreement with other peers.
 	TimeSource MedianTimeSource
 
-	// Notifications defines a callback to which notifications will be sent
-	// when various events take place.  See the documentation for
-	// Notification and NotificationType for details on the types and
-	// contents of notifications.
-	//
-	// This field can be nil if the caller is not interested in receiving
-	// notifications.
-	Notifications NotificationCallback
```

Comment:
Happy to change it if you prefer. I did it this way because it makes more sense to me to have each listener subscribe individually that have something central configure them. For example, it doesn't seem necessary to me that the server know whether the RPC server wants to respond to blockchain events. Furthermore, there would never really be contention on a read-write lock since all of the writes to notifications are made at setup time and after the server is actually started, it's all reads. But it doesn't make a big difference, and I'm fine switching to the configuration approach.